src/traces/scattergl/select.js

```diff
@@ -0,0 +1,70 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var subtypes = require('../scatter/subtypes');
+
+module.exports = function selectPoints(searchInfo, polygon) {
+    var cd = searchInfo.cd,
+        xa = searchInfo.xaxis,
+        ya = searchInfo.yaxis,
+        selection = [],
+        trace = cd[0].trace,
+        curveNumber = trace.index,
+        i,
+        di,
+        x,
+        y;
+
+    var scattergl = cd[0].glTrace;
+    var scene = cd[0].glTrace.scene;
+
+    // TODO: include lines? that would require per-segment line properties
+    var hasOnlyLines = (!subtypes.hasMarkers(trace) && !subtypes.hasText(trace));
+    if(trace.visible !== true || hasOnlyLines) return;
+
+    // filter out points by visible scatter ones
+    // var scatter2d = scattergl.scatter.instance
+
+    if(polygon === false) { // clear selection
+        for(i = 0; i < cd.length; i++) cd[i].dim = 0;
+    }
+    else {
+        for(i = 0; i < cd.length; i++) {
+            di = cd[i];
+            //FIXME: this affects performance for 1e6 points
+            x = xa.c2p(di.x);
+            y = ya.c2p(di.y);
```

Comment:
It later occurred to me that, unlike with the point picking, the lasso selection needs to be inclusive of all points, not just the visible points, so the raster based approach wouldn't work out. The rectangular selection is still fairly easy on the GPU, simply by checking whether `value === clamp(value, min, max)` as in `parcoords` but the freeform selector would need something like the even-odd raycasting, quadtree or some such...